app/services/ReviewService.scala

```diff
@@ -61,27 +69,21 @@ class ReviewService @Inject() (publishedService: PublishedService, repository: A
       case StatusPublished =>
         publishedService.save(id, info.userId, approvalProcess.meta.processCode, approvalProcess.process) map {
           case Right(_) => Right(approvalProcess)
+          case Left(DuplicateKeyError) => Left(DuplicateKeyError)
           case Left(errors) =>
             logger.error(s"Failed to publish $id - $errors")
             Left(errors)
         }
       case _ => Future.successful(Right(approvalProcess))
     }
-
     checkProcessInCorrectStateForCompletion(id, ReviewType2i) flatMap {
       case Right(ap) =>
-        reviewRepository.updateReview(id, ap.version, ReviewType2i, info.userId, info.status) flatMap {
-          case Right(()) =>
-            changeStatus(id, info.status, info.userId, ReviewType2i) flatMap {
-              case Right(_) =>
-                publishIfRequired(ap).map {
-                  case Right(_) =>
-                    ap.process
-                      .validate[Process]
-                      .fold(
-                        _ => Left(BadRequestError): RequestOutcome[AuditInfo],
-                        process => Right(AuditInfo(info.userId, ap, process))
-                      )
+        publishIfRequired(ap).flatMap {
+          case Right(ap) =>
+            reviewRepository.updateReview(id, ap.version, ReviewType2i, info.userId, info.status) flatMap {
+              case Right(()) =>
+                changeStatus(id, info.status, info.userId, ReviewType2i) map {
+                  case Right(_) => validateProcess(ap, info)
```

Comment:
It seems a bit late to validate the process here. The code will already have published the process.

Reply:
It's only really calling that method because it gets the AuditInfo back that it needs to send back to caller - the process is read from the db, so it will already have been validated when it was written to db, so maybe valudateProcess is a bad name?